engine/chain/internal/arc/broadcast.go

```diff
@@ -0,0 +1,106 @@
+package arc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	sdk "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/errors"
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/internal/ef"
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/models"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/go-resty/resty/v2"
+)
+
+// Custom ARC defined http status codes
+const (
+	StatusNotExtendedFormat             = 460
+	StatusFeeTooLow                     = 465
+	StatusCumulativeFeeValidationFailed = 473
+)
```

Comment:
Some more statuses I'd propose to handle, if you want to separate them from the general error `ErrARCUnprocessable`:
- 460 - `ErrStatusTxFormat` - if the transaction is in RawTx format and Arc cannot extend it (already here as `StatusNotExtendedFormat`)
- 463 - `ErrStatusMalformed` - if there's a problem with decoding BEEF (also if there's too many SIGOPS in scripts or if the tx is over 100 bytes)
- 465 - `ErrStatusFees` - if the fee is too low (already here)
- 467 - `ErrStatusMinedAncestorsNotFound` - if BEEF is missing any mined parents
- 469 - `ErrStatusValidatingMerkleRoots` - if merkle root validation failed (if the blocks are older than Arc retention rate - currently 28 days - then this error will not occur and validation will pass)
- 473 - `ErrStatusCumulativeFees` - cumulative fee for all blocks is too low (I don't believe you need this, unless you're using CFV)

Reply:
Actually errors `460`, `465`, `473` are currently handled separately.

The rest is handled by `default` switch-case, However, the arc-returned error message with details is wrapped and logged so no error-context is missing. 

For now, it's no use case for making it more granular. So I'll keep it as is 🥲